pkg/csi/service/vanilla/controller.go

```diff
@@ -1804,11 +1804,12 @@ func (c *controller) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequ
 				"volume capability not supported. Err: %+v", err)
 		}
 		if common.IsFileVolumeRequest(ctx, volumeCapabilities) {
-			// Error out if TopologyRequirement is provided during file volume provisioning
-			// as this is not supported yet.
-			if req.GetAccessibilityRequirements() != nil {
-				return nil, csifault.CSIInvalidArgumentFault, logger.LogNewErrorCode(log, codes.InvalidArgument,
-					"volume topology feature for file volumes is not supported.")
+			if !commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.CreateFileVolumeWithTopology) {
+				// Error out if FSS to create file volume on topology aware environment is not enabled.
+				if req.GetAccessibilityRequirements() != nil {
+					return nil, csifault.CSIInvalidArgumentFault, logger.LogNewErrorCode(log, codes.InvalidArgument,
+						"volume topology feature for file volumes is not supported.")
+				}
 			}
 			volumeType = prometheus.PrometheusFileVolumeType
 			if multivCenterCSITopologyEnabled {
```

Comment:
We need to make sure we check for vSAN file service enabled datastores in all vCenter servers when multi vCenter CSI topology is enabled.